python/cugraph/structure/graph.py

```diff
@@ -116,6 +118,81 @@ def clear(self):
         self.adjlist = None
         self.transposedadjlist = None
 
+    def add_nodes_from(self, nodes, bipartite=None, multipartite=None):
+        """
+        Add nodes information to the Graph.
+
+        Parameters
+        ----------
+        nodes : list or cudf.Series
+            The nodes of the graph to be stored. If bipartite and multipartite
+            arguments are not passed, the nodes are considered to be a list of
+            all the nodes present in the Graph.
+        bipartite : str
+            Sets the Graph as bipartite. The nodes are stored as a set of nodes
+            of the partition named as bipartite argument.
+        multipartite : str
+            Sets the Graph as multipartite. The nodes are stored as a set of
+            nodes of the partition named as multipartite argument.
+        """
```

Comment:
I am not sure that bipartite and multipartite need to be separate.  The difference is only partite count == 2 versus a different number

Looking at NetworkX:  B.add_nodes_from([1, 2, 3, 4], bipartite=0)

The add node lines specified which partite layer the nodes related to.  So you can keep track of the number of layer added and set bipartite vs multipartite accordingly.

Reply:
We can. However it might also be confusing to a user to mention the partition with the bipartite argument but enter a npartite graph